nx-X11/programs/Xserver/hw/nxagent/Keyboard.c

```diff
@@ -1904,12 +1904,14 @@ void nxagentKeycodeConversionSetup(void)
       FILE *keyboard_file;
       if (!keyboard_file_path)
       {
-        FatalError("nxagentKeycodeConversionSetup: malloc failed.");
+        free(sessionpath);
+        FatalError("malloc for keyboard file path failed.");
       }
       strcpy(keyboard_file_path, sessionpath);
       strcat(keyboard_file_path, "/keyboard");
+      free(sessionpath);
       if ((keyboard_file = fopen(keyboard_file_path, "w"))) {
-        if (drules)
+        if (drules != NULL)
```

Comment:
Changing this makes it weirdly inconsistent with the later code?

Reply:
Ping on that?

Reply:
ok, fixed

Reply:
fixed